lib/internal/modules/esm/module_job.js

```diff
@@ -151,7 +167,30 @@ class ModuleJob {
     await this.instantiate();
     const timeout = -1;
     const breakOnSigint = false;
-    await this.module.evaluate(timeout, breakOnSigint);
+    try {
+      await this.module.evaluate(timeout, breakOnSigint);
+    } catch (e) {
+      if (e.name === 'ReferenceError' &&
+          StringPrototypeEndsWith(e.message, ' is not defined') &&
+          isCommonJSGlobalLikeNotDefinedError(e.message)) {
+        e.message += ' in ES module scope';
```

Comment:
more than one of these if statements can be true at the same time. we should ensure every combination is a valid sentence or sentences.

Reply:
Well I think it is already, isn't it? Do you want me to add a test with all conditions being true at the same time?

Reply:
> we can just say `in the ES module ${url}`.

The `url` is already provided in the first line of the stack trace, is it useful to repeat it here?

Reply:
Good poin - if we definitely have an error frame for all of these then we should work to that.